src/components/buttons/z-toggle-switch/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { Component, Prop, h, Event, EventEmitter } from "@stencil/core";
+import { randomId } from "../../../utils/utils";
+
+@Component({
+  tag: "z-toggle-switch",
+  styleUrl: "styles.css",
+  shadow: false,
+  scoped: true,
+})
+export class ZToggleSwitch {
+  @Prop({ reflect: true }) disabled?: boolean = false;
+  @Prop({ reflect: true }) reverselabel?: boolean = false;
+  @Prop({ mutable: true }) checked?: boolean = false;
+  @Prop() htmlid: string = `id-${randomId()}`;
+
+  @Event() toggleClick: EventEmitter;
+  emitToggleClick(checked: boolean) {
+    this.toggleClick.emit({
+      id: this.htmlid,
+      checked: checked,
+    });
+  }
+
+  handleClick(ev) {
+    if (this.disabled) return;
+
+    this.checked = ev.target.checked;
+    this.emitToggleClick(this.checked);
+  }
+
+  render() {
+    return [
+      <input
+        id={this.htmlid}
+        type="checkbox"
+        checked={this.checked}
+        disabled={this.disabled}
+        onChange={this.handleClick.bind(this)}
+      />,
+
+      <label
```

Comment:
L'utilizzo così della label non è conferme alle specifiche html, non si riesce a gestirlo diversamente?

Specifica: _Content model: Phrasing content, but with no descendant labelable elements unless it is the element's labeled control, and no descendant label elements._